pkg/river/encoding/map.go

```diff
@@ -42,8 +42,9 @@ func (mf *mapField) convertMap(val value.Value) error {
 
 		kf.Key = key
 		mapVal, found := val.Key(key)
```

Comment:
pkg/river/token/builder ignores the found attribute and always converts the value: https://github.com/grafana/agent/blob/f77453c35f9de489faf04126305fab2cda691aa2/pkg/river/token/builder/value_tokens.go#L77

This is useful when the map has an empty value, but you'd still expect the key to show up. I think something similar should happen probably happen here.

Reply:
That could get noisy, but will add it for now and re-evaluate later.

Reply:
The only wrinkle I see is that `""` is returned as TypeNull instead of a blank string.

Reply:
I think doing a `continue` for now is ok, we need to think more about how everything behaves here. That being said, I think this is a bug: we're iterating over the list of keys from `val.Keys()`, so it's unexpected that `found` could ever be false here. 

Can you add a comment noting this as a TODO to fix the issue? We can revisit after the initial release since I don't think this edge case is a release blocker.

Reply:
Discussed and this looks like a bug/adverse behavior in `Key` func, so added a todo and will address it later.